src/System.Net.Security/src/System/Net/_SslSessionsCache.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+
+namespace System.Net.Security
+{
+    // Implements SSL session caching mechanism based on a static table of SSL credentials.
+    internal static class SslSessionsCache
+    {
+        private const int c_CheckExpiredModulo = 32;
+        private static Hashtable s_CachedCreds = new Hashtable(32);
+
+        //
+        // Uses cryptographically strong certificate thumbprint comparison.
```

Comment:
Well... the thumbprint is the SHA-1 hash of the certificate, and that's getting less cryptographically strong as time goes on :smile:.